test/query.test.js

```diff
@@ -19,4 +19,38 @@ describe('Testing query route', () => {
     expect(searchTerm.length).toBeGreaterThan(256);
     expect(res.status).toBe(400);
   });
+
+  test('Testing with empty text', async () => {
+    const res = await request(app).get('/query?filter=post');
+    expect(res.status).toBe(400);
+  });
+
+  test('Testing with empty filter', async () => {
+    const searchTerm = encodeURIComponent('I Love Telescope');
+    const res = await request(app).get(`/query?text=${searchTerm}`);
+    expect(res.status).toBe(400);
+  });
+
+  test('Testing with invalid filter', async () => {
+    const searchTerm = encodeURIComponent('I Love Telescope');
+    const res = await request(app).get(`/query?text=${searchTerm}&filter=test`);
+    expect(res.status).toBe(400);
+  });
+
+  test('Testing with valid post filter', async () => {
+    const searchTerm = encodeURIComponent('I Love Telescope');
+    const res = await request(app).get(`/query?text=${searchTerm}&filter=post`);
+    expect(res.status).toBe(200);
+  });
+
+  test('Testing with valid author filter', async () => {
+    const searchTerm = encodeURIComponent('I Love Telescope');
+    const res = await request(app).get(`/query?text=${searchTerm}&filter=author`);
+    expect(res.status).toBe(200);
+  });
+
+  test('Testing with false values', async () => {
```

Comment:
I think `empty` vs. `false` here.

Reply:
Hey @humphd , thanks for the feedback. Fixed test descriptions and squashed in one commit. Please review. Thanks.